ensime_shared/util.py

```diff
@@ -41,6 +41,11 @@ def extract_package_name(lines):
                 break
         return package
 
+def is_buffer_ensime_compatible(_vim):
```

Comment:
Please move this method to Editor, we are trying to reduce the dependency on the Vim object to only the editor module. This is the best for many reasons, but mostly for making the testing easier.